pkg/cli/cli.go

```diff
@@ -121,6 +121,55 @@ func Parse(p parser.Parser, path string, cmd *cobra.Command) (parser.Result, err
 
 	result.TestResults = append(result.TestResults, testResults)
 
+	omitStdoutForPassed, err := cmd.Flags().GetBool("omit-output-for-passed")
+	if err != nil {
+		logger.Error("Reading flag omit-output-for-passed failed with error: %v", err)
+		return result, err
+	}
+
+	if omitStdoutForPassed {
+		for idx := range result.TestResults {
+			for suiteIdx := range result.TestResults[idx].Suites {
+				for caseIdx := range result.TestResults[idx].Suites[suiteIdx].Tests {
+					if result.TestResults[idx].Suites[suiteIdx].Tests[caseIdx].State == "passed" {
+						result.TestResults[idx].Suites[suiteIdx].Tests[caseIdx].SystemErr = ""
+						result.TestResults[idx].Suites[suiteIdx].Tests[caseIdx].SystemOut = ""
+					}
+				}
+			}
+		}
```

Comment:
I think you'd need to `return` here to avoid iterating over empty strings below.

Reply:
Oh, nevermind. We still might want to apply the byte limits to the failed ones, so this is fine.